sdk-extensions/autoconfigure-spi/src/main/java/io/opentelemetry/sdk/autoconfigure/spi/internal/StructuredConfigProperties.java

```diff
@@ -0,0 +1,169 @@
+   */
+  @Nullable
+  Boolean getBoolean(String name);
+
+  /**
+   * Returns a {@link Boolean} configuration property.
+   *
+   * @return a {@link Boolean} configuration property or {@code defaultValue} if a property with
+   *     {@code name} has not been configured
+   * @throws ConfigurationException if the property is not a valid scalar boolean
+   */
+  default boolean getBoolean(String name, boolean defaultValue) {
+    return defaultIfNull(getBoolean(name), defaultValue);
+  }
+
+  /**
+   * Returns a {@link Integer} configuration property.
+   *
+   * @return null if the property has not been configured
+   * @throws ConfigurationException if the property is not a valid scalar integer
+   */
+  @Nullable
+  Integer getInt(String name);
+
+  /**
+   * Returns a {@link Integer} configuration property.
+   *
+   * @return a {@link Integer} configuration property or {@code defaultValue} if a property with
+   *     {@code name} has not been configured
+   * @throws ConfigurationException if the property is not a valid scalar integer
+   */
+  default int getInt(String name, int defaultValue) {
+    return defaultIfNull(getInt(name), defaultValue);
+  }
+
+  /**
+   * Returns a {@link Long} configuration property.
+   *
+   * @return null if the property has not been configured
+   * @throws ConfigurationException if the property is not a valid scalar long
+   */
+  @Nullable
+  Long getLong(String name);
+
+  /**
+   * Returns a {@link Long} configuration property.
+   *
+   * @return a {@link Long} configuration property or {@code defaultValue} if a property with {@code
+   *     name} has not been configured
+   * @throws ConfigurationException if the property is not a valid scalar long
+   */
+  default long getLong(String name, long defaultValue) {
+    return defaultIfNull(getLong(name), defaultValue);
+  }
+
+  /**
+   * Returns a {@link Double} configuration property.
+   *
+   * @return null if the property has not been configured
+   * @throws ConfigurationException if the property is not a valid scalar double
+   */
+  @Nullable
+  Double getDouble(String name);
+
+  /**
+   * Returns a {@link Double} configuration property.
+   *
+   * @return a {@link Double} configuration property or {@code defaultValue} if a property with
+   *     {@code name} has not been configured
+   * @throws ConfigurationException if the property is not a valid scalar double
+   */
+  default double getDouble(String name, double defaultValue) {
+    return defaultIfNull(getDouble(name), defaultValue);
+  }
+
+  /**
+   * Returns a {@link List} configuration property. Empty values will be removed. Entries which are
+   * not strings are converted to their string representation.
+   *
+   * @return a {@link List} configuration property, or null if the property has not been configured
+   * @throws ConfigurationException if the property is not a valid sequence of scalars
+   */
+  @Nullable
+  List<String> getScalarList(String name);
```

Comment:
Hmmm, do we foresee wanting lists of other non-string scalar types? I could see lists of integers, for example, being useful for defining histogram buckets or whatever. Probably plenty of other examples.  In any case, why not just call this `getStringList()`?

Reply:
Yeah this is something I wasn't sure about. Thoughts:
- Adding type-specific accessors for all the different scalar types adds more surface API surface area. We want to support booleans, integers, floats, longs, and doubles for symmetry, with each having a "get or default" method. So 10 total methods. 
- The approach here is to have a single method, since everything has an equivalent string representation, and to rely on the caller to convert string to whatever scalar type they need to access. Less surface area, but more work for callers.

Reply:
Seems fine to me. Thanks for talking it through.

Reply:
unresolving to discuss a bit more

I think I kind of prefer being explicit about the types instead of converting to String and back (unless the configuration file format intentionally defines equivalency between number lists and equivalend string lists)

in this case, the underlying json schema data types are

* String
* Boolean
* Integer (mapped to Java's Long)
* Number (mapped to Java's Double)

we probably don't need the Java Integer and Float variants, similar to how we only support Long and Double attribute values

another possible option to keep the number of methods down could be something like `List<T> getList(String name, Class<T> type)` or `List<T> getScalarList(String name, Class<T> type)`

Reply:
> List<T> getScalarList(String name, Class<T> type)

Let's do this and document that only certain known types are supported.

Reply:
Pushed a commit to reflect this convo.

Reply:
Meh I'm flip flopping. Dedicated methods for each type seems better now that I've written a single generic method:
- getDoubleList
- getLongList
- getStringList
- getBooleanList